Tutorials/Heisenberg1d.ipynb

```diff
@@ -138,70 +140,108 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 128,
+   "execution_count": 33,
    "metadata": {},
    "outputs": [],
    "source": [
-    "ma = nk.machine.Jastrow(hilbert=hi)\n",
-    "ma.init_random_parameters(seed=1, sigma=0.01)"
+    "import netket.nn as nknn\n",
+    "import flax.linen as nn\n",
+    "import jax.numpy as jnp\n",
+    "import jax\n",
+    "\n",
+    "class Jastrow(nknn.Module):\n",
+    "    @nknn.compact\n",
+    "    def __call__(self, x):\n",
+    "        x = jnp.atleast_2d(x)\n",
+    "        return jax.vmap(self.single_evaluate, in_axes=(0))(x)\n",
+    "        \n",
+    "    def single_evaluate(self, x):\n",
+    "        v_bias = self.param(\n",
+    "            \"visible_bias\", nn.initializers.normal(), (x.shape[-1],), complex\n",
+    "        )\n",
+    "\n",
+    "        J = self.param(\n",
+    "            \"kernel\", nn.initializers.normal(), (x.shape[-1],x.shape[-1]), complex\n",
+    "        )\n",
+    "\n",
+    "        return x.T@J@x + jnp.dot(x, v_bias)\n",
+    "        \n",
+    "ma = Jastrow()"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 129,
+   "execution_count": 34,
    "metadata": {},
    "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 300/300 [00:14<00:00, 20.13it/s, Energy=-39.137+0.000j ± 0.016 [σ²=0.267, R̂=0.9983]]\n"
+     ]
+    },
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
       "### Jastrow calculation\n",
-      "Has 231 parameters\n",
-      "The Jastrow calculation took 15.655055046081543 seconds\n"
+      "Has 506 parameters\n",
+      "The Jastrow calculation took 19.434601068496704 seconds\n"
      ]
     }
    ],
    "source": [
+    "# Build the sampler\n",
+    "sa = nk.sampler.MetropolisExchange(hilbert=hi,graph=g)\n",
+    "\n",
     "# Optimizer\n",
     "op = nk.optimizer.Sgd(learning_rate=0.1)\n",
     "\n",
-    "# Sampler\n",
-    "sa = nk.sampler.MetropolisExchange(machine=ma,graph=g)\n",
-    "ma.init_random_parameters(seed=12, sigma=0.01)\n",
+    "# Stochastic Reconfiguration\n",
+    "sr = nk.optimizer.SR(diag_shift=0.1)\n",
     "\n",
-    "# Stochastic reconfiguration\n",
-    "gs = nk.variational.Vmc(\n",
+    "# The variational state\n",
+    "vs = nk.variational.MCState(sa, ma, n_samples=1000)\n",
+    "\n",
+    "# The ground-state optimization loop\n",
+    "gs = nk.Vmc(\n",
     "    hamiltonian=ha,\n",
-    "    sampler=sa,\n",
     "    optimizer=op,\n",
-    "    n_samples=1000,\n",
-    "    diag_shift=0.1,\n",
-    "    method='Sr')\n",
+    "    sr=sr,\n",
+    "    variational_state=vs)\n",
     "\n",
     "start = time.time()\n",
-    "gs.run(output_prefix='Jastrow', n_iter=300)\n",
+    "gs.run(300, out='Jastrow')\n",
     "end = time.time()\n",
     "\n",
     "print('### Jastrow calculation')\n",
-    "print('Has',ma.n_par,'parameters')\n",
+    "print('Has',nk.jax.tree_size(vs.parameters),'parameters')\n",
```

Comment:
I was wondering if we could somehow still have some easy to use ```n_par``` for models? maybe ```nk.models.n_par()``` ?

Reply:
I can put it in Variational State and will internally call what you see above.

Reply:
Addressed by commit in main branch